src/main/java/org/opentripplanner/middleware/triptracker/TripInstruction.java

```diff
@@ -110,36 +114,41 @@ private static boolean isLastStep(int stepIndex, Leg expectedLeg) {
     /**
      * Provide instruction for approaching a step.
      */
-    private static String approachingStep(TripSegment activeTripSegment, StepSegment nearestStep, Leg expectedLeg) {
+    private static String approachingStep(LegSegment activeLegSegment, StepSegment nearestStep, Leg expectedLeg) {
         return getInstruction(
-            getDistance(activeTripSegment.end, nearestStep.start),
-            getStepInstruction(expectedLeg.steps, nearestStep.stepIndex)
+            getDistance(activeLegSegment.start, nearestStep.start),
+            getStep(expectedLeg.steps, nearestStep.stepIndex)
         );
     }
 
     /**
      * Determine if the trip segment is near to the destination than the step.
      */
     private static boolean isTripSegmentNearerToDestination(
-        StepSegment stepSegment,
-        TripSegment tripSegment,
+        StepSegment nearestStep,
+        LegSegment activeLegSegment,
         Coordinates destination
     ) {
-        double distanceFromLastStepToDestination = getDistance(stepSegment.start, destination);
-        double distanceFromActiveSegmentToDestination = getDistance(tripSegment.start, destination);
+        double distanceFromLastStepToDestination = getDistance(nearestStep.start, destination);
+        double distanceFromActiveSegmentToDestination = getDistance(activeLegSegment.start, destination);
         return distanceFromActiveSegmentToDestination < distanceFromLastStepToDestination;
     }
 
     /**
-     * Get instruction based distance and related step instruction.
+     * Get instruction based on distance and step instructions. e.g.
+     * <p>
+     * "CONTINUE on Langley Drive"
+     * "RIGHT on service road"
+     * "Head WEST on sidewalk approx 6 meters"
      */
-    private static String getInstruction(double distance, String stepInstruction) {
-        if (stepInstruction != null) {
-            if (distance <= IMMEDIATE_INSTRUCTION_RADIUS_DISTANCE) {
-                return IMMEDIATE_PREFIX + stepInstruction;
-            } else if (distance <= UPCOMING_INSTRUCTION_RADIUS_DISTANCE) {
-                return UPCOMING_PREFIX + stepInstruction;
-            }
```

Comment:
This was previously working. I am not able to get an instruction except for the first move after this last commit.

Reply:
Hopefully the latest updates should address this.

Reply:
@binh-dam-ibigroup is this now working again?